x-pack/test/spaces_api_integration/common/suites/create.ts

```diff
@@ -67,16 +68,12 @@ export function createTestSuiteFactory(esArchiver: any, supertest: SuperTest<any
     (describeFn: DescribeFn) =>
     (description: string, { user = {}, spaceId, tests }: CreateTestDefinition) => {
       describeFn(description, () => {
+        beforeEach(() => esArchiver.emptyKibanaIndex());
```

Comment:
Why are we cleaning before the current test instead of after the previous one? Could we continue to use `unload` since it calls `cleanSavedObjectIndices` under the hood?

https://github.com/elastic/kibana/blob/main/packages/kbn-es-archiver/src/lib/indices/delete_index_stream.ts#L32-L34

Reply:
We cleanup indices because we are reusing the existing ones, created by Kibana, and Kibana might have added some SO to it already (e.g. the default space).

We cannot rely on `unload`, cause the _delete_index_stream_ won't be triggered anymore (we deleted `mappings.json`).

Reply:
That makes sense, thanks. Though it would not be very intuitive for me as a developer writing tests that I need to cleanup every time before I import. I wonder if it would make sense to put the `cleanSavedObjectIndices` into the `load` action.

So if `load` process saved objects, then we clean the indices first. This would maintain the existing behaviour because before this PR we would have deleted an saved object index if one existed during the load.

Reply:
Fair point.

I created a PR to automatically trigger SO index cleanup whenever SO documents are detected in the `data.json`:

https://github.com/elastic/kibana/pull/159582